src/xref2/ref_tools.ml

```diff
@@ -784,7 +824,12 @@ let resolve_reference_dot_sg env ~parent_path ~parent_ref ~parent_sg name =
       Error (`Find_by_name (`Any, name))
 
 let resolve_reference_dot_page env page name =
-  L.in_page env page name >>= resolved1
+  match (L.in_page env page name, page) with
```

Comment:
That's not the usual shape of functions in this module. To make this more efficient and maintainable, it should match on the result of a single search like `resolve_reference_dot_type` below.
The correct `find` function is not in the `Find` module yet, it is currently inlined in `L.in_page` and should be extended for assets.